storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```diff
@@ -504,42 +509,45 @@ public static void validateField(String fieldName, Map conf, Class configClass)
-                        .getMethod(ConfigValidationAnnotations.ValidatorParams.VALIDATOR_CLASS).invoke(v);
-                Validator o = null;
-                Map<String, Object> params = getParamsFromAnnotation(validatorClass, v);
-                //two constructor signatures used to initialize validators.
-                //One constructor takes input a Map of arguments, the other doesn't take any arguments (default constructor)
-                //If validator has a constructor that takes a Map as an argument call that constructor
-                if(hasConstructor(clazz, Map.class)) {
-                    o  = (Validator) clazz.getConstructor(Map.class).newInstance(params);
+        try {
+            for (Annotation annotation : annotations) {
+                String type = annotation.annotationType().getName();
+                Class validatorClass = null;
+                Class<?>[] classes = ConfigValidationAnnotations.class.getDeclaredClasses();
+                //check if annotation is one of our
+                for (Class clazz : classes) {
+                    if (clazz.getName().equals(type)) {
+                        validatorClass = clazz;
+                        break;
+                    }
                 }
-                //If not call default constructor
-                else {
-                    o  = (((Class<Validator>) clazz).newInstance());
+                if (validatorClass != null) {
+                    Object v = validatorClass.cast(annotation);
+                    String key = (String) field.get(null);
+                    Class clazz = (Class) validatorClass
+                            .getMethod(ConfigValidationAnnotations.ValidatorParams.VALIDATOR_CLASS).invoke(v);
+                    Validator o = null;
+                    Map<String, Object> params = getParamsFromAnnotation(validatorClass, v);
+                    //two constructor signatures used to initialize validators.
+                    //One constructor takes input a Map of arguments, the other doesn't take any arguments (default constructor)
+                    //If validator has a constructor that takes a Map as an argument call that constructor
+                    if (hasConstructor(clazz, Map.class)) {
+                        o = (Validator) clazz.getConstructor(Map.class).newInstance(params);
+                    }
+                    //If not call default constructor
+                    else {
+                        o = (((Class<Validator>) clazz).newInstance());
+                    }
+                    o.validateField(field.getName(), conf.get(key));
                 }
-                o.validateField(field.getName(), conf.get(key));
             }
+        }  catch (NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException e) {
+            throw new IllegalStateException("ConfigValidation failed due to: " + e.getMessage() + "\n" + e.getStackTrace().toString());
```

Comment:
Would this be a operation done in an illegal state?  Maybe we could just wrap `e` in a RuntimeException?